Update QuantumESPRESSO documentation (#69)
eth-cscs/cscs-docs: docs/software/sciapps/quantumespresso.md

# ensure spack is using the uenv as upstream repository (always required)
export SPACK_SYSTEM_CONFIG_PATH=/user-environment/config
# active spack (always required)
. $SCRATCH/spack/share/spack/setup-env.sh
```

3. Create an anonymous environment for QE
```bash
spack env create -d $SCRATCH/qe-env
spack -e $SCRATCH/qe-env add quantum-espresso%nvhpc +cuda
spack -e $SCRATCH/qe-env config add packages:all:prefer:cuda_arch=90
spack -e $SCRATCH/qe-env develop -p /path/to/your/QE-src quantum-espresso@=develop
spack -e $SCRATCH/qe-env concretize -f
```
Check the output of `spack concretize -f`. All dependencies should have been picked up from spack upstream, marked eiter by a green `[^]` or `[e]`.
Next we create a local filesystem view, this instructs spack to create symlinks for binaries and libraries in a local directory `view`.
```bash
spack -e $SCRATCH/qe-env env view enable view
spack -e $SCRATCH/qe-env install
```
To recompile QE after editing the source code re-run `spack -e $SCRATCH/qe-env install`.

4. Run `pw.x` using the filesystem view generated in 3.
```bash
uenv start quantumespresso/v7.3.1
MPICH_GPU_SUPPORT_ENABLED=1 srun [...] $SCRATCH/qe-env/view/bin/pw.x < pw.in
```

> Note: The `pw.x` is linked to the uenv, it won't work without activating the uenv, also it will only work with the exact same version of the uenv. The physical installation path is in `$SCRATCH/spack`, deleting this directory will leave the anonymous spack environment created in 3. with dangling symlinks.

Reviewer: ```suggestion
!!! warning
    The `pw.x` is linked to the uenv, it won't work without activating the uenv, also it will only work with the exact same version of the uenv. 
    
!!! warning 
    The physical installation path is in `$SCRATCH/spack`, deleting this directory will leave the anonymous Spack environment created above with dangling symlinks.
```

[](){#ref-uenv-quantumespresso}
# Quantum ESPRESSO

> Quantum ESPRESSO is an integrated suite of Open-Source computer codes for electronic-structure calculations and materials modeling at the nanoscale. It is based on density-functional theory, plane waves, and pseudopotentials:

Reviewer 2: Maybe link?
```suggestion
[Quantum ESPRESSO](https://www.quantum-espresso.org/) is an integrated suite of Open-Source computer codes for electronic-structure calculations and materials modeling at the nanoscale. It is based on density-functional theory, plane waves, and pseudopotentials:
```

> PWscf (Plane-Wave Self-Consistent Field)
> FPMD (First Principles Molecular Dynamics)
> CP (Car-Parrinello)

Reviewer: ```suggestion
* PWscf (Plane-Wave Self-Consistent Field)
* FPMD (First Principles Molecular Dynamics)
* CP (Car-Parrinello)
```
Maybe it's worth using the executables names here? Not sure...

## ALPS (GH200)
### Setup

> Download the uenv image for QuantumESPRESSO:

Reviewer 2: ```suggestion
Download the uenv image for Quantum ESPRESSO:
```
?

### List available images

```bash
uenv image find quantumespresso
```

### Pull the image of interest

```
uenv image pull quantumespresso/v7.4:v2

> ```

Reviewer: I think this might be redundant with the UENV documentation, and we don't want to have to change this if the `uenv` interface changes. In CP2K I added the following after the description of the software at the very top.

https://github.com/eth-cscs/cscs-docs/blob/dca32a4077d5f89f1dd9e08e5cc97f65d4d5a9cd/docs/software/sciapps/cp2k.md?plain=1#L13-L16

What do you think?

> QuantumESPRESSO can be compiled from source using the above uenv. The procedure is described in https://eth-cscs.github.io/alps-uenv/uenv-qe/#building-a-custom-version.

Reviewer: I think the idea is to port over that documentation here. Would you mind moving it? I'd put it after the "how to run" one (as for the CP2K docs).

Author: Ok, good point I'll add the how to build

> QuantumESPRESSO can be compiled from source using the above uenv. The procedure is described in https://eth-cscs.github.io/alps-uenv/uenv-qe/#building-a-custom-version.

Reviewer 2: ```suggestion
Quantum ESPRESSO can be compiled from source using the above uenv. The procedure is described in https://eth-cscs.github.io/alps-uenv/uenv-qe/#building-a-custom-version.
```
?

### How to run

> The following sbatch script can be used as a template.

Reviewer 2: Might be useful to link to https://eth-cscs.github.io/cscs-docs/running/slurm/#one-rank-per-gpu? I think a full example here is good, but I think it's also good to link to the more generic documentation should the user need/want to modify things.

Author: I've added the link to one rank and also for multiple ranks per gpu.
